lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java

```diff
@@ -0,0 +1,367 @@
+                        MAX_ATTEMPTS);
+
+                eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_USER_REDIRECTED_ATTEMPTS_OVER_MAX);
+
+                APIGatewayProxyResponseEvent responseEvent =
+                        passportSuccessResponseEvent(sessionItem);
+
+                // Use the completed OK exit sequence
+                return lambdaCompletedOK(responseEvent);
+            }
+
+            PassportFormData passportFormData = parsePassportFormRequest(input.getBody());
+            eventProbe.counterMetric(FORM_DATA_PARSE_PASS);
+
+            DocumentDataVerificationResult documentDataVerificationResult =
+                    documentDataVerificationService.verifyData(
+                            passportFormData, sessionItem, requestHeaders);
+
+            saveAttempt(sessionItem, passportFormData, documentDataVerificationResult);
+
+            boolean canRetry =
+                    determineVerificationRetryStatus(
+                            sessionItem, documentDataVerificationResult, MAX_ATTEMPTS);
+            LOGGER.info("CanRetry {}", canRetry);
+
+            APIGatewayProxyResponseEvent responseEvent =
+                    determineExitResponseEvent(sessionItem, canRetry);
+
+            // Use the completed OK exit sequence
+            return lambdaCompletedOK(responseEvent);
+        } catch (SessionNotFoundException e) {
+
+            String customOAuth2ErrorDescription = SESSION_NOT_FOUND.getMessage();
+            LOGGER.error(customOAuth2ErrorDescription);
+            eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
+
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatusCode.FORBIDDEN,
+                    new CommonExpressOAuthError(
+                            OAuth2Error.ACCESS_DENIED, customOAuth2ErrorDescription));
+        } catch (SessionExpiredException e) {
+
+            String customOAuth2ErrorDescription = SESSION_EXPIRED.getMessage();
+            LOGGER.error(customOAuth2ErrorDescription);
+            eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
+
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatusCode.FORBIDDEN,
+                    new CommonExpressOAuthError(
+                            OAuth2Error.ACCESS_DENIED, customOAuth2ErrorDescription));
+        } catch (OAuthHttpResponseExceptionWithErrorBody e) {
+            eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    e.getStatusCode(), // Status Code determined by throw location
+                    new CommonExpressOAuthError(OAuth2Error.SERVER_ERROR));
+        } catch (Exception e) {
+            eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
+            // Cannot log e, due to possibility of PII
+            LOGGER.error("Check Passport Unhandled Exception");
```

Comment:
We should pickup masking as soon as we can as I dont like not getting the full exceptions